Remove build methods from SolverConfig and PlannerBenchmarkConfig (#302)
kiegroup/optaplanner-website: download/upgradeRecipe/upgradeRecipe8.adoc

If you were also passing a `ClassLoader`, pass it to both `SolverConfig.createFromXmlResource()` and `SolverFactory.create()`.

[.upgrade-recipe-minor.upgrade-recipe-public-api]
=== `SolverConfig`: `buildSolver()` removed

> The method `SolverConfig.buildSolver()` was an inner method that should not have been used.

Reviewer: very passive voice: "should not have been"

```suggestion
The method `SolverConfig.buildSolver()` is an inner method that doesn't belong in the public API.
```

Instead, use `SolverFactory.buildSolver()`.

Before in `*.java`:

[source,java]
----
SolverConfig solverConfig = SolverConfig.createFromXmlResource(".../mySolverConfig.xml");
...
Solver<MySolution> solver = solverConfig.buildSolver();
----

After in `*.java`:

[source,java]
----
SolverConfig solverConfig = SolverConfig.createFromXmlResource(".../mySolverConfig.xml");
...
SolverFactory<MySolution> solverFactory = SolverFactory.create(solverConfig);
Solver<MySolution> solver = solverFactory.buildSolver();
----

[.upgrade-recipe-minor.upgrade-recipe-public-api]
=== `PlannerBenchmarkConfig`: `buildPlannerBenchmark()` removed

> The method `PlannerBenchmarkConfig.buildPlannerBenchmark()` was an inner method that should not have been used.

Reviewer: ```suggestion
The method `PlannerBenchmarkConfig.buildPlannerBenchmark()` is an inner method that doesn't belong in the public API.
```
